Gems/GradientSignal/Code/Source/UI/GradientPreviewWidget.cpp

```diff
@@ -33,6 +64,26 @@ namespace GradientSignal
         return size();
     }
 
+    void GradientPreviewWidget::enterEvent(QEvent* event)
+    {
+        QWidget::enterEvent(event);
+
+        if (m_popoutButton)
+        {
+            m_popoutButton->show();
+        }
+    }
+
+    void GradientPreviewWidget::leaveEvent(QEvent* event)
+    {
+        QWidget::leaveEvent(event);
+
+        if (m_popoutButton)
+        {
+            m_popoutButton->hide();
+        }
+    }
```

Comment:
nit: would the behavior be more consistent if the calls were in reverse order on leave?

```
    void GradientPreviewWidget::enterEvent(QEvent* event)
    {
        QWidget::enterEvent(event);

        if (m_popoutButton)
        {
            m_popoutButton->show();
        }
    }

    void GradientPreviewWidget::leaveEvent(QEvent* event)
    {
        if (m_popoutButton)
        {
            m_popoutButton->hide();
        }

        QWidget::leaveEvent(event);
    }

```

Reply:
I typically like to have the custom logic afterwards in case any part of the custom logic depended on the `QWidget` internal state (e.g. if you override `resizeEvent` and your custom logic used `size()`).